### DISCUSSION TITLE
Getting WifiClientSecure to use PSRAM

### DISCUSSION THREAD

**Discussion by codegrue** [espressif/arduino-esp32#9305] [Q&A]
Has anyone gotten this to work and if so can you please post a code example? I have scoured the internet for possible solutions but there is nothing consistent and nothing seems to work. The best I found is WifiClientSecure should just use PSRAM if it's present but it is always using the main heap in all my testing.

I am using platformio and the arduino framework so the menuconfig function is not available.

**Comment by ToyboxZach** [comment]
I believe you have to change up how you use platromio to make it compile the espressif framework as well:

I have this:
```
[env]
platform = espressif32
framework = arduino, espidf

platform_packages =
    platformio/framework-espidf@3.40405
    platformio/framework-arduinoespressif32@3.20011.230801
```

And then I define sdkconfig.defaults and it generates it correctly

```

#
# SPI RAM config
#
CONFIG_ESP32_SPIRAM_SUPPORT=y
CONFIG_SPIRAM_TYPE_AUTO=y
# CONFIG_SPIRAM_TYPE_ESPPSRAM16 is not set
# CONFIG_SPIRAM_TYPE_ESPPSRAM32 is not set
# CONFIG_SPIRAM_TYPE_ESPPSRAM64 is not set
CONFIG_SPIRAM_SIZE=-1
CONFIG_SPIRAM_SPEED_80M=y
CONFIG_ESPTOOLPY_FLASHFREQ_80M=y
# CONFIG_SPIRAM_SPEED_40M is not set
CONFIG_SPIRAM=y
CONFIG_SPIRAM_BOOT_INIT=y
# CONFIG_SPIRAM_IGNORE_NOTFOUND is not set
# CONFIG_SPIRAM_USE_MEMMAP is not set
# CONFIG_SPIRAM_USE_CAPS_ALLOC is not set
CONFIG_SPIRAM_USE_MALLOC=y
CONFIG_SPIRAM_MEMTEST=y
CONFIG_SPIRAM_MALLOC_ALWAYSINTERNAL=4096
CONFIG_SPIRAM_MALLOC_RESERVE_INTERNAL=65536
CONFIG_SPIRAM_ALLOW_BSS_SEG_EXTERNAL_MEMORY=y
```

I don't remember if we had to do extra set up or not, but it is possible to get it.
